platforms/chibios/BLACKPILL_STM32_F401/ld/STM32F401xE_tinyuf2.ld

```diff
@@ -0,0 +1,88 @@
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+/*
+ * STM32F401xE memory setup.
+ */
+MEMORY
+{
+    flash0 (rx) : org = 0x08000000 + 64k, len = 512k - 64k /* tinyuf2 bootloader requires app to be located at 64k offset for this MCU */
+    flash1 (rx) : org = 0x00000000, len = 0
+    flash2 (rx) : org = 0x00000000, len = 0
+    flash3 (rx) : org = 0x00000000, len = 0
+    flash4 (rx) : org = 0x00000000, len = 0
+    flash5 (rx) : org = 0x00000000, len = 0
+    flash6 (rx) : org = 0x00000000, len = 0
+    flash7 (rx) : org = 0x00000000, len = 0
+    ram0   (wx) : org = 0x20000000, len = 96k
+    ram1   (wx) : org = 0x00000000, len = 0
+    ram2   (wx) : org = 0x00000000, len = 0
+    ram3   (wx) : org = 0x00000000, len = 0
+    ram4   (wx) : org = 0x00000000, len = 0
+    ram5   (wx) : org = 0x00000000, len = 0
+    ram6   (wx) : org = 0x00000000, len = 0
+    ram7   (wx) : org = 0x00000000, len = 0
+}
+
+/* For each data/text section two region are defined, a virtual region
+   and a load region (_LMA suffix).*/
+
+/* Flash region to be used for exception vectors.*/
+REGION_ALIAS("VECTORS_FLASH", flash0);
+REGION_ALIAS("VECTORS_FLASH_LMA", flash0);
+
+/* Flash region to be used for constructors and destructors.*/
+REGION_ALIAS("XTORS_FLASH", flash0);
+REGION_ALIAS("XTORS_FLASH_LMA", flash0);
+
+/* Flash region to be used for code text.*/
+REGION_ALIAS("TEXT_FLASH", flash0);
+REGION_ALIAS("TEXT_FLASH_LMA", flash0);
+
+/* Flash region to be used for read only data.*/
+REGION_ALIAS("RODATA_FLASH", flash0);
+REGION_ALIAS("RODATA_FLASH_LMA", flash0);
+
+/* Flash region to be used for various.*/
+REGION_ALIAS("VARIOUS_FLASH", flash0);
+REGION_ALIAS("VARIOUS_FLASH_LMA", flash0);
+
+/* Flash region to be used for RAM(n) initialization data.*/
+REGION_ALIAS("RAM_INIT_FLASH_LMA", flash0);
+
+/* RAM region to be used for Main stack. This stack accommodates the processing
+   of all exceptions and interrupts.*/
+REGION_ALIAS("MAIN_STACK_RAM", ram0);
+
+/* RAM region to be used for the process stack. This is the stack used by
+   the main() function.*/
+REGION_ALIAS("PROCESS_STACK_RAM", ram0);
+
+/* RAM region to be used for data segment.*/
+REGION_ALIAS("DATA_RAM", ram0);
+REGION_ALIAS("DATA_RAM_LMA", flash0);
+
+/* RAM region to be used for BSS segment.*/
+REGION_ALIAS("BSS_RAM", ram0);
+
+/* RAM region to be used for the default heap.*/
+REGION_ALIAS("HEAP_RAM", ram0);
+
+/* Generic rules inclusion.*/
+INCLUDE rules.ld
+
+/* TinyUF2 bootloader reset support */
+_board_dfu_dbl_tap = ORIGIN(ram0) + 64k - 4; /* this is based off the linker file for tinyuf2 */
```

Comment:
I think this will be correct, given that they use the same firmware for the blackpill boards, both f401 and f411... however, can you see if you can confirm on the tinyuf2 source that this is indeed where it's looking?

The F401xE variants have 96kB RAM, just confirming.

Reply:
check this readme https://github.com/adafruit/tinyuf2/blob/master/ports/stm32f4/README.md
Is that what you are asking? I am sorry I was just copying from your original code and it works in flashing my 401 keyboard.

Reply:
Okay, so what I was getting at was that in order to enter bootloader from software, you need to write a known value to a well-defined location in RAM before resetting. Upon reset, the bootloader consults that location in RAM to check if it needs to stay in bootloader mode, or to run the user-mode firmware such as QMK.

This is the definition used for STM32F4:
https://github.com/adafruit/tinyuf2/blob/1ef93470263b26ec56d8815a540512184ed74971/ports/stm32f4/linker/stm32f4.ld#L35-L49

You'll note that there's a definition of a location called `_board_dfu_dbl_tap` which matches the naming in this PR. The calculation in the `ld` file needed to match the calculation in the link above... and as far as I can tell, as tinyuf2 uses the same script for both F401 and F411, nothing should require changing -- what's in the PR seems like it should work.

Still, would you mind confirming that adding `RESET` to your keymap does indeed reset the board into the bootloader mode?

Reply:
This is to confirm that both double tapping the reset switch in the board as well as pressing the RESET keycode in the keymap gets to the bootloader mode (i.e. new usb drive is shown) for both f401 and f411 keyboards.